contracts/Removal.sol

```diff
@@ -22,38 +23,25 @@ contract Removal is
   ERC1155SupplyUpgradeable
 {
   using SafeMathUpgradeable for uint256;
-
-  struct Vintage {
-    address supplier;
-    uint16 vintage;
-    // todo: location
-    // todo: methodology
-    // todo: supplier name
-  }
+  using RemovalUtils for uint256;
 
   struct BatchMintRemovalsData {
     address marketAddress;
     bool list;
   }
 
-  mapping(uint256 => Vintage) private _vintages;
-  uint256 private _latestTokenId;
+  uint256 private _tokenIdCounter;
   string public name; // todo why did I add this
+  mapping(uint256 => uint256) public indexToTokenId;
+  mapping(uint256 => bool) private _tokenIdExists;
 
   function initialize() public virtual initializer {
     super.initialize("https://nori.com/api/removal/{id}.json");
     __ERC1155Supply_init_unchained();
-    _latestTokenId = 0;
+    _tokenIdCounter = 0;
```

Comment:
remind me, is there still value in a nested tokenid? can we also just encode this? A bit dusty on our previosus conversations.

Reply:
as discussed, left a todo next to this to reconsider whether there's a more appropriate data structure to use to meet our needs on this one, but we're going to leave the sequential index ---> removal id map there for now